### PR TITLE
extract commands for testability

### PR DESCRIPTION
**Pull request by jtarchie**
I’m making this PR before I make subsequent PRs. Mainly to ensure that this is “alright”. I’d like to limit the need to compile an entire binary to test the code. Mainly for readability.

This PR does maintain the interface of the resource thought. Using stdin/stdout/stderr and arguments.

### CODE REVIEW

**Comment by jtarchie** [comment]
Will fix sign off shortly. Sorry.

**Comment by xtreme-sameer-vohra** [comment]
Thanks for the PR. Sorry for the delay in getting to it, taking a look now

File: go.sum [hunk 1]
```diff
@@ -23,50 +23,6 @@ github.com/alecthomas/units v0.0.0-20151022065526-2efee857e7cf/go.mod h1:ybxpYRF
 github.com/apache/thrift v0.12.0/go.mod h1:cp2SuWMxlEZw2r+iP2GNCdIi4C1qmUzdZFSVb+bacwQ=
 github.com/armon/consul-api v0.0.0-20180202201655-eb2c6b5be1b6/go.mod h1:grANhF5doyWs3UAsr3K4I6qtAmlQcZDesFNEHPZAzj8=
 github.com/aws/aws-sdk-go v1.15.90/go.mod h1:es1KtYUFs7le0xQ3rOihkuoVD90z7D0fR2Qm4S00/gU=
-github.com/aws/aws-sdk-go v1.19.1 h1:8kOP0/XGJwXIFlYoD1DAtA39cAjc15Iv/QiDMKitD9U=
```
Comment (xtreme-sameer-vohra) [inline go.sum:26]: 👍 tidying these up

**Comment by xtreme-sameer-vohra** [comment]
> I’m making this PR before I make subsequent PRs. Mainly to ensure that this is “alright”. I’d like to limit the need to compile an entire binary to test the code. Mainly for readability.
> 
> This PR does maintain the interface of the resource thought. Using stdin/stdout/stderr and arguments.

PRs to improve/enhance are quite welcome. 
If you're planning to submit changes soon, might make sense to keep an 👀  on these #230 #214

**Comment by jtarchie** [comment]
Well thank you for merging, but those other PRs look like some large ones. This probably just affected their whole rebasing strategy.

**Pull request merged**